src/foolscap/banana.py

```diff
@@ -346,8 +346,11 @@ def pushSlicer(self, slicer, obj):
         # methods which are *not* generators.
 
         itr = slicer.slice(topSlicer.streamable, self)
-        next = iter(itr).next
-
+        if six.PY2:
```

Comment:
Are we aiming for py2+py3 compatibility? Or just py3? I originally wanted to get both, but at this point I think it'd be fine to abandon py2.

Reply:
We want at least one version of Tahoe-LAFS that supports both so users can transition..

Reply:
Aiming for py2+3 - Yes. Hence this exercise of making tests pass in Py2 while gently introducing 3 compatibility.